### PR TITLE
Replace check runs with workflow output commands

### PR DESCRIPTION
**Pull request by sheck**
Details in the commit messages.

Links that might be helpful:
* [Workflow output command docs](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions)
* [Comment that helped me to discover this as a solution](https://github.com/orgs/community/discussions/14891#discussioncomment-3953207)
* [First slack thread](https://pco.slack.com/archives/CSHSCUYPR/p1666202510300699)
* [Second slack thread](https://pco.slack.com/archives/C046SCNDVRB/p1666287072600989)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203228724523988
  - https://app.asana.com/0/0/1204129897385069

### CODE REVIEW

File: action/action.rb [hunk 1]
```diff
@@ -105,10 +109,11 @@ def report_offense?(offense, change_ranges:)
...
   puts e.backtrace.inspect
-  resp = check_run.error(message: e.message)
-  p resp
-  p resp.json
+  abort("::error:: #{e.message}")
 else
-  check_run.update(annotations: annotations)
   ActionUtils.set_output("issuesCount", annotations.count)
+  annotations.each do |note|
+    puts note.to_output_command
```
Comment (danielma) [inline action/action.rb:116]: Do you think it's worth using `ActionUtils` to format out annotation output?

**Review by danielma — APPROVED** [review]
I really like this solution. A few things in particular:

- No more special permissions
- Extremely easy to test (if we wanted to) since it's just string output

I appreciate very much the cons you listed. That's very helpful to think through this as a solution. I feel good about it going forward.

Thinking even further into the future, given the state of this repo, I think we could get away with writing it all in JS now, since we're shelling out to Rubocop and have no HTTP interaction anymore

**Comment by sheck** [comment]
@danielma 

> Extremely easy to test (if we wanted to) since it's just string output

That was my thought too. I tried getting testing with Act working but ran into so many problems with the test images available and the ruby versions included.

> Thinking even further into the future, given the state of this repo, I think we could get away with writing it all in JS now, since we're shelling out to Rubocop and have no HTTP interaction anymore

I also had a similar thought while working on this. Testing this would be much easier, and could probably stub out an expected response from rubocop. Also, I think once we do that, we could start extracting/sharing utilities for getting relevant changed lines too.

File: action/action.rb (hunk 1 above)
Comment (sheck) [reply action/action.rb:116]: Currently ActionUtils just has a utility to echo debug commands and to set the output of the workflow entirely (like with annotationCount) not to "output" these types of commands. We could move the logic (or extract a portion) from `Annotation#to_output_command` but I think I would prefer to leave it the way it is — I'm only like 3/10 on that though.

**Pull request merged**